edx/edx-enterprise
# enable continue button on enterprise course enrollment page

@asadiqbal08 @saleem-latif @douglashall 

**Description:** Enable "Continue" button on enterprise course enrollment page (enterprise landing page).

**JIRA:** [ENT-379](https://openedx.atlassian.net/browse/ENT-379)

**Dependencies:**
* ~~[ENT-348](https://openedx.atlassian.net/browse/ENT-348) https://github.com/edx/edx-enterprise/pull/109~~ **Merged**

**Merge deadline:** 16 June, 2017

**Installation instructions:** Install `edx-enterprise` from this branch `zub/ENT-379-ent-landing-page-enable-continue-button` in `edx-platform`.

**Testing instructions:**
1. Go to enterprise customers django admin at `admin/enterprise/enterprisecustomer/`
2. Add enterprise customer with logo and with/without (2 different flows) consent disabled and link it with an `idp`.
3. Create/publish a course in studio
4. Login as enterprise learner to view the enterprise course enrollment page (enterprise landing page) for the enterprise course by accessing the enterprise course enrollment url with enterprise customer uuid from step no 2, e.g: http://zubair.localhost:8000/enterprise/72416e52-8c77-4860-9584-15e5b06220fb/course/course-v1:edX+TEST_01+2016_R1/enroll/
5. On enterprise course enrollment page verify the `Continue` button flows:
    1. **User Selects Audit Track**
        * if DSC needed, display DSC. After successful DSC, enroll learner and go to Dashboard
        * if DSC not needed, enroll learner and go to Dashboard
    2. **User Selects Verified/Prof Ed Track**
    **`WITH ENROLLMENT CODE`**
        1. If DSC needed, display DSC. After successful DSC,  display Basket with redeem code
        2. IF DSC not needed, display Basket with redeem code
    **`WITH DISCOUNT CODE`**
        1. If DSC needed, display DSC. After successful DSC, display Basket with discount applied
        2. IF DSC not needed, display Basket with discount applied
    **`NO SEAT ENTITLEMENT`**
        1. If DSC needed, display DSC. After successful DSC, display Basket
        2. IF DSC not needed, display Basket

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
